src/core/algorithms/od/fastod/attribute_pair.cpp

```diff
@@ -0,0 +1,18 @@
+#include "attribute_pair.h"
+
+#include <string>
+
+namespace algos::fastod {
+
+AttributePair::AttributePair(AttributeSet::size_type left, AttributeSet::size_type right)
+    : left(left), right(right) {}
+
+std::string AttributePair::ToString() const {
+    return "{ " + std::to_string(left + 1) + ", " + std::to_string(right + 1) + " }";
```

Comment:
We use column numbering starting from one because we use this strategy in order dependency classes. We do this in order to maintain a consistent numbering style in the output. In order dependency classes we use this strategy because this strategy is used in the Java implementation of the algorithm with which we are comparing.